CHANGELOG.md

```diff
@@ -1,5 +1,173 @@
+
+Since its introdcuction, the cache would essentially be reset each time AiiDA or any of the plugin packages would be updated, since the version of these packages were included in the calculation of the node hashes.
+This was originally done out of precaution to err on the safe-side and limit the possibility of false-positives in cache hits.
+However, this strategy has turned out to be unnecessarily cautious and severely limited the effectiveness of caching.
+
+The package version information is no longer included in the hash and therefore no longer impacts the caching.
+This change does now make it possible for false positives if the implementation of a `CalcJob` or `Parser` plugin changes signficantly.
+Therefore, a mechanism is introduced to give control to these plugins to effectively reset the cache of existing nodes.
+Please refer to the [documentation on controlling caching](https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/provenance/caching.html#calculation-jobs-and-parsers) for more details.
+
+### Programmatic syntax for query builder filters and projections
+
+In the `QueryBuilder`, fields to filter on or project always had to be provided with strings:
+```python
+QueryBuilder().append(Node, filters={'label': 'some-label'}, project=['id', 'ctime'])
+```
+and it is not always trivial to know what fields exist that _can_ be filtered on and can be projected.
+In addition, there was a discrepancy for some fields, most notably the `pk` property, which had to be converted to `id` in the query builder syntax.
+
+These limitations have been solved as each class in AiiDA's ORM now defines the `fields` property, which allows to discover these fields programmatically.
+The example above would convert to:
+```python
+QueryBuilder().append(Node, filters={'label': Node.fields.label}, project=[Node.fields.pk, Node.fields.ctime])
+```
+The `fields` property provides tab-completion allowing easy discovery of available fields for an ORM class in IDEs and interactive shells.
+The fields also allow to express logical conditions programmatically and more.
+For more details, please refer to the [documentation on programmatic field syntax](https://aiida.readthedocs.io/projects/aiida-core/en/latest/howto/query.html#programmatic-syntax-for-filters).
+
+Data plugins can also define custom fields, adding on top of the fields inherited from their base class(es).
+The [documentation on data plugin fields](https://aiida.readthedocs.io/projects/aiida-core/en/latest/topics/data_types.html#fields) provides more information, but the API is currently in beta and guaranteed to be changed in an upcoming version.
+It is therefore recommended for plugin developers to hold off making use of this new API.
+
+### Automated profile storage backups
+
+A generic mechanism has been implemented to allow easily backing up the data of a profile.
```

Comment:
Is there more documentation that we can link to here? Specifically, it is not clear to me how to restore backups generated in this way. (and this is not explained in the `verdi storage backup` help either)

Reply:
Done